src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -660,13 +660,6 @@ private void finish(@NonNull Result r, @CheckForNull Throwable t) {
                 listener = null;
             }
             saveWithoutFailing(true);
-            Timer.get().submit(() -> {
-                try {
-                    getParent().logRotate();
-                } catch (Exception x) {
-                    LOGGER.log(Level.WARNING, "failed to perform log rotation after " + this, x);
-                }
-            });
```

Comment:
Originally made asynchronous in https://github.com/jenkinsci/workflow-job-plugin/commit/63fdbe8c2d0f74f97e8f3b92cc566e2fff52bdf3. Note though that since https://github.com/jenkinsci/jenkins/pull/4368, Jenkins core was _also_ calling `Job.logRotate` synchronously from `Run.onEndBuilding` -> `RunListener.onFinalized` (at least with the default global build discarder configuration), so this call was redundant.

With this PR only Jenkins core calls `Job.logRotate`, and it does so synchronously. We could make that call asynchronous if there are any concerns.

Reply:
(#70)